File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cs
internal class Program
{
    static List<int>[] graph;
    static bool[] visited;

    static void Main()
    {
        int[] a = Array.ConvertAll(Console.ReadLine()!.Split(), int.Parse);

        graph = new List<int>[a[0]];
        visited = new bool[a[0]];
        for (int i = 0; i < a[0]; i++)
            graph[i] = new List<int>();

        for (int i = 0; i < a[1]; i++)
        {
            int[] aa = Array.ConvertAll(Console.ReadLine()!.Split(), int.Parse);

            graph[aa[0] - 1].Add(aa[1] - 1);
            graph[aa[1] - 1].Add(aa[0] - 1);
        }

        for (int i = 0; i < a[0]; i++)
            graph[i].Sort();

        DFS(a[2] - 1);

        Console.Write("\n");

        visited = new bool[a[0]];

        BFS(a[2] - 1);
    }

    static void DFS(int v)
    {
        visited[v] = true;
        Console.Write(v + 1 + " ");

        foreach (int value in graph[v])
        {
            if (!visited[value])
                DFS(value);
        }
    }

    static void BFS(int v)
    {
        Queue<int> queue = new Queue<int>();
        queue.Enqueue(v);
        visited[v] = true;

        while (queue.Count > 0)
        {
            int a = queue.Dequeue();
            Console.Write(a + 1 + " ");

            foreach (int value in graph[a])
            {
                if (!visited[value])
                {
                    visited[value] = true;
                    queue.Enqueue(value);
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>

using namespace std;

int N, M, V;
vector<int> graph[1001];
bool visited[1001];

void dfs(int node) {
    cout << node << " ";
    visited[node] = true;

    for (int i = 0; i < graph[node].size(); i++)
    {
        if (!visited[graph[node][i]])
        {
            dfs(graph[node][i]);
        }
    }
}

void bfs(int start) {
    queue<int> q;
    fill(visited, visited + 1001, false);

    q.push(start);
    visited[start] = true;

    while (!q.empty())
    {
        int x = q.front();
        cout << x << " ";
        q.pop();


        for (int i = 0; i < graph[x].size(); i++)
        {
            if (!visited[graph[x][i]])
            {
                q.push(graph[x][i]);
                visited[graph[x][i]] = true;
            }
        }
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    cin >> N >> M >> V;

    for (int i = 0; i < M; i++) {
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for (int i = 1; i <= N; i++) {
        sort(graph[i].begin(), graph[i].end());
    }

    dfs(V);
    cout << "\n";

    fill(visited, visited + 1001, false);
    bfs(V);

    return 0;
}